harvester/config.go

```diff
@@ -0,0 +1,242 @@
+package harvester
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+)
+
+func UnmarshalDiskInfo(data []byte) (DiskInfo, error) {
+	var r DiskInfo
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *DiskInfo) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+type DiskInfo struct {
+	Disks []Disk `json:"disks"`
+}
+
+type Disk struct {
+	ImageName        string `json:"imageName"`
+	StorageClassName string `json:"storageClassName"`
+
+	Size      int `json:"size"`
```

Comment:
Question:
Is the disk size unit be `Gb` by default?, it could be too big in Edge scenarios.
Is it possible to add an "unit" to support `Gb`, `Mb`, `Tb` ? or still use string to represent the disk size